### PR TITLE
SEAB-5132: Reimplement MuteForSuccessfulTests equivalent for Junit 5

### PR DESCRIPTION
**Pull request by svonworl**
**Description**
This PR introduces a new Junit 5 extension `MuteForSuccessfulTests` that, upon any test failure, prints the captured output from any SystemStubs system streams, such as `SystemOut` or `SystemErr`, to stdout.  Otherwise, the default behavior occurs, which is that said streams capture output but do not print it.

I've updated all the tests that use `SystemOut` or `SystemErr` so they use the `MuteForSuccessfulTests` extension.

For lack of a better place, I put the extension into dockstore-common.  It is included in the test jar, which is a dependency of the integration test build.  Please check out the corresponding mods to the pom to confirm that they are ok.

**Review Instructions**
Find a CircleCI test run that incorporates this change that failed, and make sure the its output to stdout/stderr is included in the logs.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-5132

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

File: dockstore-common/src/test/java/io/dockstore/common/MuteForSuccessfulTests.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    public void testFailed(ExtensionContext context, Throwable cause) {
+        try {
+            Object test = context.getRequiredTestInstance();
+            // Iterate over the fields of the test object.
+            for (Field field: test.getClass().getDeclaredFields()) {
+                // Skip any field that doesn't extend SystemStreamBase,
+                // thus avoiding a potential security/access exception on an irrelevant private field.
+                if (SystemStreamBase.class.isAssignableFrom(field.getType())) {
+                    // Extract the "stream" field value.
+                    field.setAccessible(true);
```
Comment (denis-yuen) [inline dockstore-common/src/test/java/io/dockstore/common/MuteForSuccessfulTests.java:46]: Curious if PMD's suggestion about @SuppressWarnings works.

**Review by denis-yuen — COMMENTED** [review]
Curious about sonar cloud output, looks good in principle, request re-review when passing

**Review by coverbeck — APPROVED** [review]
Lots of files! Nice fix.

**Review by denis-yuen — APPROVED** [review]
Nice!

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/ServiceIT.java [hunk 2]
```diff
@@ -71,20 +72,21 @@
 import uk.org.webcompere.systemstubs.jupiter.SystemStubsExtension;
 import uk.org.webcompere.systemstubs.stream.SystemErr;
 import uk.org.webcompere.systemstubs.stream.SystemOut;
-import uk.org.webcompere.systemstubs.stream.output.NoopStream;
 
 /**
  * @author dyuen
  */
 @ExtendWith(SystemStubsExtension.class)
+@ExtendWith(MuteForSuccessfulTests.class)
```
Comment (fhembroff) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/ServiceIT.java:80]: I think you may have copied this one in twice.

Comment (fhembroff) [reply dockstore-integration-testing/src/test/java/io/dockstore/webservice/ServiceIT.java:80]: On line 80 and 82

**Review by fhembroff — APPROVED** [review]

**Pull request merged**